main/src/library/DelayMap.flix

```diff
@@ -342,6 +342,16 @@ namespace DelayMap {
     pub def foldRight(f: (v, b) -> b & ef, s: b, m: DelayMap[k, v]): b & ef =
         foldRightWithKey((_, v, acc) -> f(v, acc), s, m)
 
+    ///
+    /// Applies `f` to a start value `s` and all values in `m` going from right to left.
+    /// The function f must be pure.
+    ///
+    /// That is, the result is of the form: `f(x1, ...lazy f(xn-1, lazy f(xn, s))...)`.
+    ///
+    @Experimental
+    pub def foldRightL(f: (v, Lazy[b]) -> b, s: b, m: DelayMap[k, v]): b =
+        foldRightWithKey((_, v, acc) -> f(v, lazy acc), s, m)
```

Comment:
I am not smart enough to understand if this is actually lazy. I worry that calling foldRightWithKey will actually force the entire computation, but I am not sure, and you maybe right.

Reply:
Can you explain/argue why using foldRight is okay?

Reply:
I have also the same doubt, could you please explain me better the semantics of lazy and force keywords to understand this better?

Reply:
However my intuition is that the value calculated by `f` is still lazily evaluated but the values taken from `m` are all calculated inside `foldRightWithKey`, but this should be ok since `f` takes lazily only the accumulator.

Maybe we want also another operator that also does not forces the values from the delayed map